### PR TITLE
ref(deletion): Split up cleanup method

### PR DESCRIPTION
**Pull request by armenzg**
The clean-up method was very long and hard to follow. This should make it a bit easier to read.

### CODE REVIEW

File: src/sentry/runner/commands/cleanup.py [hunk 1]
```diff
@@ -400,6 +289,202 @@ def is_filtered(model: type[Model]) -> bool:
...
+        (UserReport, "date_added", None),
+        (GroupEmailThread, "date", None),
+        (RuleFireHistory, "date_added", None),
+        (NotificationMessage, "date_added", None),
+    ] + additional_bulk_query_deletes
+
+    return BULK_QUERY_DELETES
+
+
+def running_bulk_query_deletes(bulk_query_deletes, is_filtered, days, project, project_id):
```
Comment (markstory) [inline src/sentry/runner/commands/cleanup.py:426]:
```suggestion
def run_bulk_query_deletes(bulk_query_deletes, is_filtered, days, project, project_id):
```

🤷

File: src/sentry/runner/commands/cleanup.py [hunk 2]
```diff
@@ -200,121 +195,30 @@ def is_filtered(model: type[Model]) -> bool:
...
         if SiloMode.get_current_mode() != SiloMode.CONTROL:
             if project:
-                click.echo("Bulk NodeStore deletion not available for project selection", err=True)
-                project_id = get_project(project)
-                # These models span across projects, so let's skip them
-                DELETES.remove((models.ArtifactBundle, "date_added", "date_added"))
-                if project_id is None:
-                    click.echo("Error: Project not found", err=True)
-                    raise click.Abort()
+                remove_cross_project_models()
```
Comment (markstory) [inline src/sentry/runner/commands/cleanup.py:213]: Does this need a parameter?

File: src/sentry/runner/commands/cleanup.py [hunk 3]
```diff
@@ -400,6 +289,202 @@ def is_filtered(model: type[Model]) -> bool:
...
+    for entry in settings.ADDITIONAL_BULK_QUERY_DELETES:
+        app_name, model_name = entry[0].split(".")
+        model_tp = apps.get_model(app_name, model_name)
+        additional_bulk_query_deletes.append((model_tp, entry[1], entry[2]))
+
+    BULK_QUERY_DELETES = [
+        (UserReport, "date_added", None),
+        (GroupEmailThread, "date", None),
+        (RuleFireHistory, "date_added", None),
+        (NotificationMessage, "date_added", None),
```
Comment (markstory) [inline src/sentry/runner/commands/cleanup.py:420]: We'll get a conflict here.

File: src/sentry/runner/commands/cleanup.py (hunk 2 above)
Comment (armenzg) [reply src/sentry/runner/commands/cleanup.py:213]: Yes. I'm fixing it locally.

File: src/sentry/runner/commands/cleanup.py (hunk 3 above)
Comment (armenzg) [reply src/sentry/runner/commands/cleanup.py:420]: I think you just swapped the two; right?

I have changed it locally.

**Comment by armenzg** [comment]
@markstory Could you please review it? I will fix the conflicts afterward.

**Review by markstory — APPROVED** [review]
Looks good to me.

File: src/sentry/runner/commands/cleanup.py [hunk 4]
```diff
@@ -398,6 +290,214 @@ def is_filtered(model: type[Model]) -> bool:
...
+    # (model, datetime_field, order_by)
+    additional_bulk_query_deletes = []
+    for entry in settings.ADDITIONAL_BULK_QUERY_DELETES:
+        app_name, model_name = entry[0].split(".")
+        model_tp = apps.get_model(app_name, model_name)
+        additional_bulk_query_deletes.append((model_tp, entry[1], entry[2]))
+
+    BULK_QUERY_DELETES = [
+        (UserReport, "date_added", None),
+        (GroupEmailThread, "date", None),
```
Comment (armenzg) [inline src/sentry/runner/commands/cleanup.py:423]: I've made sure to port the changes from this PR:
https://github.com/getsentry/sentry/pull/77289/files#diff-af1cde82f065157a87af5b9c019c93a9e93aeb22661494449dbca1628e3cf306R212

File: src/sentry/runner/commands/cleanup.py [hunk 5]
```diff
@@ -398,6 +290,214 @@ def is_filtered(model: type[Model]) -> bool:
...
+
+    # Deletions that use the `deletions` code path (which handles their child relations)
+    # (model, datetime_field, order_by)
+    return [
+        (EventAttachment, "date_added", "date_added"),
+        (ReplayRecordingSegment, "date_added", "date_added"),
+        (ArtifactBundle, "date_added", "date_added"),
+        (MonitorCheckIn, "date_added", "date_added"),
+        (GroupRuleStatus, "date_added", "date_added"),
+        (RuleFireHistory, "date_added", "date_added"),
```
Comment (armenzg) [inline src/sentry/runner/commands/cleanup.py:372]: I've made sure to add this as well:
https://github.com/getsentry/sentry/pull/77289/files#diff-af1cde82f065157a87af5b9c019c93a9e93aeb22661494449dbca1628e3cf306R223
